Topics/Development_Process/Build_Requirements/Introduction_to_make_and_Makefiles.md

```diff
@@ -0,0 +1,228 @@
+- ```make``` make encourages you to record your code: The modular nature of Makefiles encourages you to record each step you make, enabling you and others to reliably reproduce the entire process. You are essentially mapping out a directed acyclic graph of your project with Makefiles.
+- ```make``` allows for easy and reliable CI: By providing a Makefile with common ```make``` targets like ```build```, ```compile```, ```lint```, and ```test```, you can start writing your project  Your CI will execute your ```make``` targets, deleting your generated files and rebuilding from scratch to test. You can then utilize other more modern dependency managers or script runners under the hood.
+- ```make``` functions off of timestamps: timestamps are built into approximately every filesystem, so you don't need to store extra metadata or do any further checks.
+
+#### Disadvantages
+- ```make```'s syntax is clunky and hard to learn: The [full manual](https://www.gnu.org/software/make/manual/html_node/index.html) is 183 pages and is not very pretty and not that intuitive, including whitespace sensitivity, where indentation errors are difficult to spot by simply looking. The built-in functions and constructs lack some complex logical capabilities and programming features that can be difficult to express in a Makefile.
+- ```make``` functions off of timestamps: Rebuilding due to changes in timestamps can be unreliable/undesirable and inefficient when content is not changed.
+- ```make``` platform independency writing can be difficult: Handling all the cases of which compilers and tools to use in cases for different platforms can be tedious and challenging.
+- ```make``` error messages are not always clear: It can be difficult to identify and fix issues from the built-in error messages alone, especially in larger complex build systems.
+- ```make``` doesn't easily address external dependencies: Managing source files dependencies is what ```make``` was built for, but third party library and package dependency management would require additional effort or external tools.
+
+## Additional Resources
+- The official GNU ```make``` documentation: https://www.gnu.org/software/make/manual/html_node/index.html
+- A more detailed tutorial on Makefiles: https://github.com/Nuno-Jesus/Make-A-Make
+- A simple Makefile example: https://github.com/remonbonbon/makefile-example/tree/master
+- A more advanced Makefile managed project: https://github.com/skwee357/wrap-os/tree/master
+
+## Footnotes
+
+[^1]: The prefix for recipes can be changed by changing the .RECIPEPREFIX variable, for example the following is a valid makefile:
+```
+.RECIPEPREFIX = ~
+hello:
+~ @echo hello world
+```
+[^2]: If you want to call make on a file with a different name, use the ```-f``` option, e.g. calling ```make``` on a file named foo.mk:
+```
+make -f foo.mk
+```
+
```

Comment:
I think these notes would be more useful at the bottom of the relevant section than at the bottom of the entire guide.